### DISCUSSION TITLE
Render form components individually

### DISCUSSION THREAD

**Discussion by 4m1n0s** [filamentphp/filament#2685] [Help]
Hey,

i'm asking if there is a way to render form components individually? instead of using $this->form

**Comment by khanakia** [comment]
How to use form inside widget and then render that widget on page ?
```
protected function getHeaderWidgets(): array
{
    return [
        MyFormWidget::class
    ];
}
```

**Reply by danharrin** [reply]
The widget is the Livewire component

https://filamentphp.com/docs/2.x/forms/getting-started

**Reply by khanakia** [reply]
@danharrin I just tried it does not works getting the error 

Because before my class was extending the components `class EditPost extends Component implements Forms\Contracts\HasForms` but i changed it to `class EditPost extends Widget implements Forms\Contracts\HasForms`

```
Livewire \ Exceptions \ 
ComponentNotFoundException
PHP 8.1.8
9.22.1
Unable to find component: []
```

Where do i register the  widget as i am using Simple Custom Page `class Settings extends Page{}` so i do not find anywhere to register the widget i am not using the any resource for this page.

**Reply by danharrin** [reply]
Call `Livewire::component('edit-post-widget', EditPost::class)` from a service provider
